### PR TITLE
A Couple of Changes

### PR DESCRIPTION
**Pull request by alexandersanford**
Significantly refactored the data structures used by our code base. Also added an Entry class that enables more direct manipulation of each entry (row in database). There's a lot of changes so please look through this over the next 24 hours and let me know what you think.

### CODE REVIEW

**Comment by Alejandro-Rojas** [comment]
if we are using this it has to be tested on the springboot framework that has to be merged still.  It should work since everything else we  program wise we had ran fine on it, when I tested it.

**Review by kevvo15 — COMMENTED** [review]
For the most part, this program functions very good. There are a few minor logical errors that need to be fixed in BuyerEvent that relate to the updateQuantity() method. There are also a few errors that occur when attempting to create a new entry in main.java. Both these errors seem to stem from the fact that the new "Entry" class tends to produce null objects when called in methods, resulting in null pointer exceptions and other unexpected runtime errors. Aside from this and a few other minor issues (inventory file and buyer event file that are read in program are not located anywhere in the program; easy fix tho) I still managed to get the program running. If this can be merged without having conflicts with the new spring boot version of our program, then I think this is a good basis for moving on forward.

File: src/main/java/com/dreamteam/database/Entry.java [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	}
+	
+	public boolean subtractQuantity(int increment) {
+		return setQuantity(getQuantity() - increment);
+	}
+	
+	private boolean setQuantity(int quantity) {
+		if(quantity >= 0) {
+			this.quantity = quantity;
+			System.out
```
Comment (jfreda21) [inline src/main/java/com/dreamteam/database/Entry.java:71]: ought to be one line, unnecessary white space

**Review by JonathanGrant92 — COMMENTED** [review]
As the main.java class has the main method for the program, the Database.java class probably should not have this method.

**Review by JonathanGrant92 — COMMENTED** [review]
In Database.java, parseEntry() should probably be merged into the create() method.

I believe the toString() method is causing the entire hashmap to be printed to the console, where ever it is called. I don't know why we should want that behavior.

File: src/main/java/com/dreamteam/database/Entry.java (hunk 1 above)
Comment (JonathanGrant92) [reply src/main/java/com/dreamteam/database/Entry.java:71]: You're right. System.out.println(String) should probably by on one line.

**Review by Alejandro-Rojas — APPROVED** [review]
Looks good there were some bug, but they were small that we can fix

**Pull request closed without merging**